daemon/src/taker_cfd.rs

```diff
@@ -132,39 +128,24 @@ where
         Ok(())
     }
 
-    async fn handle_propose_settlement(
-        &mut self,
-        msg: ProposeSettlement,
-        ctx: &mut xtra::Context<Self>,
-    ) -> Result<()> {
+    async fn handle_propose_settlement(&mut self, msg: ProposeSettlement) -> Result<()> {
         let ProposeSettlement {
             order_id,
             current_price,
         } = msg;
 
-        let disconnected = self
-            .collab_settlement_actors
-            .get_disconnected(order_id)
-            .with_context(|| format!("Settlement for order {} is already in progress", order_id))?;
-
-        let mut conn = self.db.acquire().await?;
-        let cfd = load_cfd(order_id, &mut conn).await?;
-
-        let this = ctx
-            .address()
-            .expect("actor to be able to give address to itself");
-        let (addr, fut) = collab_settlement_taker::Actor::new(
-            cfd,
-            self.projection_actor.clone(),
-            this,
+        let (_, fut) = collab_settlement_taker::Actor::new(
```

Comment:
Are you sure we can just drop the address right away? Won't that reduce the ref-count to zero? I think we may have to keep it alive at least until the end of this function so that the `started` function runs (where we clone the address and send it to the connection::Actor, god that is complicated ....).

Reply:
We actually need to save it, because the ref-count reaches zero before the spawned task clones the address (unless we sleep here which would be ludicrous).

Reply:
It's very unfortunate because this actor doesn't care about these ephemeral actors after spawning them.

Reply:
What you could do is:

```rust
self.tasks.add(async move {
	let _addr = address;
	fut.await;
});
```

To keep the address alive for as long as the context is running. I am not sure if that is elegant or hacky :joy:

Reply:
I considered this, but it's a bit suspicious...